presto-tests/src/test/java/com/facebook/presto/tests/TestQueryManager.java

```diff
@@ -226,4 +227,45 @@ public void testQueryOutputSizeExceeded()
             assertEquals(queryInfo.getErrorCode(), EXCEEDED_OUTPUT_SIZE_LIMIT.toErrorCode());
         }
     }
+
+    @Test
+    public void testQueryCountMetrics()
+            throws Exception
+    {
+        DispatchManager dispatchManager = queryRunner.getCoordinator().getDispatchManager();
+        // Create a total of 4 queries to test concurrency limit and ensure Queued Queries
+        createQueries(dispatchManager, 3);
+
+        // This is to ensure the 4th Query we want to monitor is Queued
+        while (dispatchManager.getStats().getRunningQueries() != 3) {
+            Thread.sleep(25);
```

Comment:
How was 25 chosen? Is it possible the query runs faster than 25msec and this loop never ends?